src/Microsoft.ServiceFabric.Actors/Diagnostics/ActorFrameworkEventSource.cs

```diff
@@ -11,17 +11,20 @@
 namespace Microsoft.ServiceFabric.Actors.Diagnostics
 {
     // REMARKS:
-    // When you apply EventAttribute attribute to an ETW event method defined on an EventSource-derived class,
+    // 1. When you apply EventAttribute attribute to an ETW event method defined on an EventSource-derived class,
     // you must call the WriteEvent method on the base class, passing the event ID, followed by the same
     // arguments as the defined method is passed. Details at:
     // https://msdn.microsoft.com/en-us/library/system.diagnostics.tracing.eventattribute(v=vs.110).aspx
+    // 2. This class is not intended to be inherited even though it is not sealed. 
+    // We kept this class unsealed and it's methods virtual in order to enable mocking of this class and verification of 
+    // method calls in the unit tests.
     [EventSource(Name = "Microsoft-ServiceFabric-Actors", LocalizationResources = "Microsoft.ServiceFabric.Actors.SR", Guid = "0e1ec353-9f02-55d7-fbb8-f3857458acbd")]
-    internal sealed class ActorFrameworkEventSource : ServiceFabricEventSource
+    internal class ActorFrameworkEventSource : ServiceFabricEventSource
     {
         internal static ActorFrameworkEventSource Writer { get; private set; } = new ActorFrameworkEventSource();
 
         [NonEvent]
-        internal void ReplicaChangeRoleToPrimary(ServiceContext serviceContext)
+        internal virtual void ReplicaChangeRoleToPrimary(ServiceContext serviceContext)
```

Comment:
I don't remember if I suggested before and you probably already know that we can observe event method calls using EventListener without making the "NonEvent" methods virtual. However, verifying the high-level event methods with mocks is much more concise so overall, I think this approach is *simpler*.